### PR TITLE
Fixed hard fault on rover_pos_control stop

### PR DESCRIPTION
**Pull request by ItsTimmy**
**Describe problem solved by the proposed pull request**
When running the command `rover_pos_control stop`, it hard-faults.

**Test data / coverage**
Stopped and started `rover_pos_control` several times to make sure it did not hard-fault.

**Describe your preferred solution**
An object was being double-deleted. That has been fixed.

### CODE REVIEW

**Comment by dagar** [comment]
It's probably worth switching this over entirely to ModuleBase to fix these problems. Take a look at the current FW position or attitude controllers as an example.

**Comment by bkueng** [comment]
Or look at the template: https://github.com/PX4/Firmware/tree/master/src/templates/module. Also for the parameters.

File: src/modules/rover_pos_control/RoverPositionControl.cpp [hunk 1]
```diff
@@ -510,95 +440,89 @@ RoverPositionControl::task_main()
...
+
+### Examples
+CLI usage example:
+$ rover_pos_control start
+$ rover_pos_control status
+$ rover_pos_control stop
+
+)DESCR_STR");
+
+	PRINT_MODULE_USAGE_NAME("rover_pos_control", "command");
```
Comment (bkueng) [inline src/modules/rover_pos_control/RoverPositionControl.cpp]:
```suggestion
	PRINT_MODULE_USAGE_NAME("rover_pos_control", "controller");
```

File: src/modules/rover_pos_control/RoverPositionControl.cpp [hunk 2]
```diff
@@ -240,13 +182,13 @@ RoverPositionControl::control_position(const matrix::Vector2f &current_position,
 
 		matrix::Vector2f ground_speed_2d(ground_speed);
 
-		float mission_throttle = _parameters.throttle_cruise;
+		float mission_throttle = _param_throttle_cruise.get();
 
 		/* Just control the throttle */
-		if (_parameters.speed_control_mode == 1) {
+		if (_param_speed_control_mode .get() == 1) {
```
Comment (bkueng) [inline src/modules/rover_pos_control/RoverPositionControl.cpp]:
```suggestion
		if (_param_speed_control_mode.get() == 1) {
```

File: src/modules/rover_pos_control/RoverPositionControl.cpp [hunk 3]
```diff
@@ -510,95 +440,89 @@ RoverPositionControl::task_main()
...
+	PRINT_MODULE_USAGE_COMMAND("start")
+	PRINT_MODULE_USAGE_DEFAULT_COMMANDS();
+
+	return 0;
+}
 
-	warnx("unrecognized command");
-	return 1;
+int
+RoverPositionControl::print_status()
```
Comment (bkueng) [inline src/modules/rover_pos_control/RoverPositionControl.cpp]: You don't need to override this if you don't print additional information

File: src/modules/rover_pos_control/RoverPositionControl.hpp [hunk 4]
```diff
@@ -66,38 +66,43 @@
...
+	/** @see ModuleBase::print_status() */
+	int print_status() override;
 
 private:
 	orb_advert_t	_pos_ctrl_status_pub{nullptr};		/**< navigation capabilities publication */
 	orb_advert_t    _actuator_controls_pub{nullptr};	/**< actuator controls publication */
 
-	bool		_task_should_exit{false};		/**< if true, sensor task should exit */
+
 	bool		_task_running{false};			/**< if true, task is running in its mainloop */
```
Comment (bkueng) [inline src/modules/rover_pos_control/RoverPositionControl.hpp]: You can remove this as well

File: src/modules/rover_pos_control/RoverPositionControl.cpp [hunk 5]
```diff
@@ -503,102 +428,85 @@ RoverPositionControl::task_main()
 	orb_unsubscribe(_pos_sp_triplet_sub);
 	orb_unsubscribe(_vehicle_attitude_sub);
 
-	_task_running = false;
-
 	warnx("exiting.\n");
```
Comment (bkueng) [inline src/modules/rover_pos_control/RoverPositionControl.cpp]: fyi you can remove that as well, and `warnx` is deprecated, `PX4_{WARN,INFO,ERR}` is used instead.

**Pull request merged**